serializer-deserializer/src/main/java/com/amazonaws/services/schemaregistry/deserializers/GlueSchemaRegistryDeserializationFacade.java

```diff
@@ -97,6 +97,15 @@ public GlueSchemaRegistryDeserializationFacade(@NonNull GlueSchemaRegistryConfig
         this.cache = GlueSchemaRegistryDeserializerCache.getInstance(glueSchemaRegistryConfiguration);
     }
 
+    /**
+     * Method to override user-agent app name for the de-serializer.
+     * This overrides the previously set value in GlueSchemaRegistryConfiguration.
+     * @param name AppName
+     */
+    public void overrideUserAgentApp(String name) {
+        this.glueSchemaRegistryConfiguration.setUserAgentApp(name);
```

Comment:
why allow overriding ?

Reply:
This is required for KCL to set this explicitly. KCL doesn't have access to `GlueSchemaRegistryConfiguration`, so can't set it via configuration. [Ref](https://github.com/awslabs/amazon-kinesis-client/blob/master/amazon-kinesis-client/src/main/java/software/amazon/kinesis/schemaregistry/SchemaRegistryDecoder.java)